docs/DeveloperGuide.md

```diff
@@ -1,11 +1,118 @@
+* <code> LoginManager </code> The LoginManager provides the logic to verify the user login details
+* <code> LoginUi </code> The LoginUi defines the display and gets input from the user. LoginUi extends the functionality of the Ui class
+* <code> LoginInfo </code> The LoginInfo contains the atrributes and methods of each LoginInfo object from a particular user logging into MojoHR application
+* <code> LoginInfoFileManager </code> The LoginInfoFileManager is responsible for the logic that handles the storage and retrieval of login information
+
+### Email Component
+The email component consists of classes that are involved with all the emails
+![Email Component Class Diagram](UML diagrams/EmailClassDiagram.png)
+The **Login Class Diagram** give above shows how the classes in the Email Component interact with each other
+
+The Email Component consists of 8 components.
+* <code> Email </code> is the parent class to the other email classes and contains all the information that is required in an email
+* <code> Archive </code> inherits from Email
+* <code> Deleted </code> inherits from Email
+* <code> Draft </code> inherits from Email
+* <code> Inbox </code> inherits from Email
+* <code> Junk </code> inherits from Email
+* <code> Sent </code> inherits from Email
+* <code> EmailManager </code> is responsible for the logic of retrieving Emails as a whole
+
+## Implementation
+
+### ListCommand 
+There are seven variations of the <code> list </code> command.
+1. List by All email(s)
+2. List by Archived email(s) 
+3. List by Deleted email(s)
+4. List by Draft(s) email(s) 
+5. List by Inbox email(s)
+6. List by Junk email(s)
+7. List by Sent email(s)
+
+The sequence diagram shows how the <code> list(type) </code> operation work. 
+![ListCommand Sequence Diagram](Sequence Diagrams/ListCommand.png)
+
+### TagCommand
+`TagCommand` allows the user to tag a specific email with the a number of tags.
+1. `TagCommand` first finds out which email the user chose to be tagged by searching through the listedEmailList for the index given.
+2. `TagCommand` will list the available tags that the users will choose from by calling `printTag()` on `Ui`
+3. `TagCommand` then pass the user's input to the `parser` to extract the index
+4. `TagCommand` sets the tag to the email through `EmailManager`
+5. `TagCommand` updates the hard disk through `Storage`.
+
+The sequence diagram shows how the <code> list(type) </code> operation work.
+![Tag Sequence Diagram](Sequence Diagrams/TagSequenceDiagram.png)
+
+###ComposeCommand
+When the `compose` command is entered, the software will prompt user to enter the necessary details and save them to draft.
+
+The details needed are receiver email, subject and content. The time the draft was composed would be saved automatically.
+
+The sequence diagram roughly shows how the `compose` operation works.
+![ComposeCommand Sequence Diagram](Sequence Diagrams/ComposeCommand.png)
```

Comment:
![image](https://user-images.githubusercontent.com/24218271/113240295-0689f980-92df-11eb-89c1-9068f5b6fac0.png)
Same for this, there should be dotted return arrows from all function calls. Also, is `ComposeCommand` deleted after `printEmailDraft()` ends?